### DISCUSSION TITLE
Is there a GlobalTracerProvider() method which will give me a singleton tracerprovider instance?

### DISCUSSION THREAD

**Discussion by varkey98** [open-telemetry/opentelemetry-dotnet#5877] [Q&A] [answered]
I have seen this in opentelemetry-go and opentelemetry-cpp where we have a global api giving a tracer provider instance. Not sure if this is part of the OTel API, but was wondering if we have anything similar in opentelemetry-dotnet as well?
function in go: https://github.com/open-telemetry/opentelemetry-go/blob/main/trace.go#L29

**Comment by cijothomas** [comment, marked as answer]
The global is required usually to obtain Tracer (ActivitySource), Meter instances. OTel .NET allows you to do create them directly using ctors, and use AddSource, AddMeter to "connect" them to the overall sdk. 

In short, you don't need it in OTel .NET. Check the examples in this repo and you'll see that they don't use/need global apis.

**Reply by varkey98** [reply]
Understood, I wanted to do a tracer provider flush. I don't want to DI the tracer provider instance and was thinking that if there is a global API, I could just use that to get the instance and do a flush. What would be an alternative here?

**Reply by cijothomas** [reply]
https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/trace#tracerprovider-management DI is recommended, if your app already has DI, but not mandatory.

**Reply by varkey98** [reply]
Got it, thanks.
